polkadot/runtime/polkadot/Cargo.toml

```diff
@@ -311,9 +313,13 @@ try-runtime = [
 # A feature that should be enabled when the runtime should be build for on-chain
 # deployment. This will disable stuff that shouldn't be part of the on-chain wasm
 # to make it smaller like logging for example.
-on-chain-release-build = [ "sp-api/disable-logging" ]
+on-chain-release-build = [ "disable-genesis-builder", "sp-api/disable-logging" ]
 
 # Set timing constants (e.g. session period) to faster versions to speed up testing.
 fast-runtime = []
 
 runtime-metrics = [ "runtime-parachains/runtime-metrics", "sp-io/with-tracing" ]
+
+# When enabled, the GenesisBuilder API will not be supported, GenesisConfig shall be
+# stripped from the final binary
+disable-genesis-builder = []
```

Comment:
Not sure we need them. Please remove them.

Reply:
If the size is not a concern, then we can keep serde + genesis builder within binary. On the other hand it is not used on-chain. Less code, maybe better - I mean some audits, attack vectors, etc...

Anyway, I am also not sure, so I'll remove it.

Reply:
7d2b3f1798